core/templates/components/question-directives/question-editor/question-editor.component.ts

```diff
@@ -13,253 +13,258 @@
-        QuestionUpdateService.setQuestionInapplicableSkillMisconceptionIds(
-          ctrl.question, displayedValue);
-      };
-
-      ctrl.showMarkAllAudioAsNeedingUpdateModalIfRequired = function(
-          contentIds) {
-        var state = ctrl.question.getStateData();
-        var recordedVoiceovers = state.recordedVoiceovers;
-        var writtenTranslations = state.writtenTranslations;
-        var updateQuestion = _updateQuestion;
-
-        const shouldPrompt = contentIds.some(
-          contentId =>
-            recordedVoiceovers.hasUnflaggedVoiceovers(contentId));
-        if (shouldPrompt) {
-          NgbModal.open(
-            MarkAllAudioAndTranslationsAsNeedingUpdateModalComponent, {
-              backdrop: 'static'
-            }).result.then(function() {
-            updateQuestion(function() {
-              contentIds.forEach(contentId => {
-                if (recordedVoiceovers.hasUnflaggedVoiceovers(contentId)) {
-                  recordedVoiceovers.markAllVoiceoversAsNeedingUpdate(
-                    contentId);
-                }
-                if (
-                  writtenTranslations.hasUnflaggedWrittenTranslations(
-                    contentId)
-                ) {
-                  writtenTranslations.markAllTranslationsAsNeedingUpdate(
-                    contentId);
-                }
-              });
-            });
-          }, function() {
-            // This callback is triggered when the Cancel button is
-            // clicked. No further action is needed.
+import { StateEditorService } from 'components/state-editor/state-editor-properties-services/state-editor.service';
+import { StateInteractionIdService } from 'components/state-editor/state-editor-properties-services/state-interaction-id.service';
+import { EditabilityService } from 'services/editability.service';
+import { SolutionValidityService } from 'pages/exploration-editor-page/editor-tab/services/solution-validity.service';
+import { LoaderService } from 'services/loader.service';
+import { QuestionUpdateService } from 'domain/question/question-update.service';
+import cloneDeep from 'lodash/cloneDeep';
+import { SubtitledHtml } from 'domain/exploration/subtitled-html.model';
+import { Hint } from 'domain/exploration/HintObjectFactory';
+import { Solution } from 'domain/exploration/SolutionObjectFactory';
+import { InteractionCustomizationArgs } from 'interactions/customization-args-defs';
+import { Outcome } from 'domain/exploration/OutcomeObjectFactory';
+import { AnswerGroup } from 'domain/exploration/AnswerGroupObjectFactory';
+import { Question } from 'domain/question/QuestionObjectFactory';
+import { State } from 'domain/state/StateObjectFactory';
+
+@Component({
+  selector: 'oppia-question-editor',
+  templateUrl: './question-editor.component.html'
+})
+export class QuestionEditorComponent implements OnInit, OnDestroy {
+  @Input() questionId: unknown;
+  @Input() misconceptionsBySkill: object;
```

Comment:
`object` is too generic. Use a interface instead.

Reply:
done